Assets/Character/AaronTestAssets/PlayerState.cs

```diff
@@ -0,0 +1,74 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class PlayerState : MonoBehaviour {
+
+
+    private Vector3 scale;
+    protected float direction = 1;
+    protected int facing = 1;
+    protected bool grounded = false;
+    protected new string name;
+    protected SpriteRenderer sr;
+    protected IdleState idle;
```

Comment:
Each state needs instances of other states to switch, so I'm not 100% sure how else to implement this. Maybe I could store them in an array in player state and have it switch from there?